### PR TITLE
Add crude tests for printing nil in various ways

### PR DESCRIPTION
**Pull request by timothywalter**
As requested in issue NLua/NLua#353 though all tests seem to pass

### CODE REVIEW

File: tests/src/TestTypes/PrintOverrides.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            string output = "";
+
+            if (arguments != null)
+            {
+                foreach (object argument in arguments)
+                {
+                    if (argument == null)
+                    {
+                        output += SYMBOL_PRINT_CONCAT + "nil";
```
Comment (viniciusjarina) [inline tests/src/TestTypes/PrintOverrides.cs]: I would use a StringBuilder here

File: tests/src/LuaTests.cs [hunk 2]
```diff
@@ -1766,6 +1766,111 @@ public void TestPrintChars()
...
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintManyNilLastNumberParamsDetour()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.RegisterFunction("print", PrintOverrides.GetPrintDetourMethodInfo());
```
Comment (viniciusjarina) [inline tests/src/LuaTests.cs]: Also, I think there are easier ways to binding a method without all the reflection stuff, take a look in the other tests that we have.

Comment (timothywalter) [reply tests/src/LuaTests.cs]: Thanks for the heads-up! After fooling around a bit more I realized the only reason I needed all the reflection stuff was because I mad the method private (which in my head was a great idea since only Lua would be using the method). For simplicity's sake I've changed it to public and simplified the registration

File: tests/src/LuaTests.cs [hunk 3]
```diff
@@ -1766,6 +1766,111 @@ public void TestPrintChars()
...
+            }
+        }
+
+        [Test]
+        public void TestPrintSingleNil()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.DoString(@"print(nil)");
+                Assert.IsTrue(true);
```
Comment (viniciusjarina) [inline tests/src/LuaTests.cs]: One more thing. 
What if we make Print retur the string and use the return in the assertion instead always true?
This way you could compare the expected result with the actual result avoiding future regressions

Comment (timothywalter) [reply tests/src/LuaTests.cs]: I've made Print return the string and we now compare the first chunk returned from Lua with the expected result. Is this what you meant + is it now in a way you'd like to see it?

Thanks for the feedback, learning a lot!
